cinema/views.py

```diff
@@ -87,6 +115,8 @@
         )
     )
     serializer_class = MovieSessionSerializer
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
 
     def get_queryset(self):
         date = self.request.query_params.get("date")
```

Comment:
Consider adding error handling for the date parsing to avoid potential exceptions if the date format is incorrect. This will make your code more robust.